client/src/components/notification/NotificationBox.tsx

```diff
@@ -0,0 +1,137 @@
+import { NotificationData } from "@/fetches/notification/notification.type";
+import { useEffect, useState } from "react"
+import { useAuth } from "@/contexts/AuthContext";
+import { EventSourcePolyfill } from "event-source-polyfill";
+import { server } from "@/fetches/common/axios";
+import { ResponseWithData, ResponseWithoutData } from "@/fetches/common/response.type";
+import BellIcon from "../svgs/BellIcon";
+import NotificationItem from "./NotificationItem";
+
+export default function NotificationBox() {
+  const [showBox, setShowBox] = useState(false);
+  const [notificationArr, setNotificationArr] = useState<NotificationData[]>([]);
+  const { auth } = useAuth();
+
+  const onDeleteAll = async () => { 
+    if (notificationArr.length > 0) {
+      const response = await server.delete<ResponseWithoutData>('/notifications');
+    
+      if (response.success) {
+        setNotificationArr([]);
+        setShowBox(false);
+      } else {
+        alert(response.message);
+      }
+    }
+  };
+
+  const onDeleteOne = async (notificationId: number) => { 
+    const response = await server.delete<ResponseWithoutData>(`/notifications/${notificationId}`);
+    
+    if (response.success) {
+      setNotificationArr((prevList: NotificationData[]) => {
+        return prevList.filter((item) => item.id !== notificationId);
+      });
+    } else {
+      alert(response.message);
+    }
+  };
+
+  useEffect(() => {
+    if (auth.userId) {
+
+      const getNotifications = async () => {
+        const response = await server.get<ResponseWithData<NotificationData[]>>('/notifications');
+      
+        if (response.success) {
+          setNotificationArr(response.data);
+        } else {
+          alert(response.message)
+          setNotificationArr([]);
+        }
+      };
+      
+      getNotifications();
+
+      const eventSource = new EventSourcePolyfill("http://localhost:8080/api/v1/notifications/subscribe", {
+        headers: {
+          Authorization: 'Bearer ' + auth.accessToken
+        },
+      });
+
+      eventSource.addEventListener('message', (event) => {
+        if(event.data !== 'Connected') {
+          const newNotification: NotificationData = JSON.parse(event.data);
+          
+          if (newNotification.title !== '') {
+            setNotificationArr((prevList) => [newNotification, ...prevList]);
+          }
+        }
```

Comment:
```suggestion
        if(event.data === 'Connected') return;
        const newNotification: NotificationData = JSON.parse(event.data);
          
        if (newNotification.title === '') return;
        setNotificationArr((prevList) => [newNotification, ...prevList]);
```

return 을 이용해서 뎁스를 줄여 가족성을 높일 수 있을 것같아요!